File: Array/Remove Duplicates from Sorted Array/solution1.py
from typing import List


class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        write = 0
        for read in range(1, len(nums)):
            if nums[read] != nums[write]:
                write += 1
                nums[write] = nums[read]
        return write + 1


s = Solution()
print(s.removeDuplicates([1, 1, 2, 3, 3, 4, 4, 5]))

File: Array/Array Waveform/solution1.go
package main

import (
	"fmt"
)

func WaveArray(arr []int) {
	l := len(arr)
	if l <= 1 {
		return
	}
	for i := range arr {
		if i%2 == 0 {
			continue
		}
		if arr[i] > arr[i-1] {
			arr[i], arr[i-1] = arr[i-1], arr[i]
		}

		if i+1 < l && arr[i] > arr[i+1] {
			arr[i], arr[i+1] = arr[i+1], arr[i]
		}
	}

}

func main() {
	nums := []int{8, 1, 2, 3, 4, 5, 6, 4, 2}
	WaveArray(nums)
	fmt.Println(nums)
}
